src/openzaak/components/zaken/api/filters.py

```diff
@@ -29,6 +31,73 @@
     ZaakObject,
     ZaakVerzoek,
 )
+from .serializers.authentication_context import (
+    DigiDLevelOfAssurance,
+    eHerkenningLevelOfAssurance,
+)
+
+# custom filter to show cases for authorizee and representee
+MACHTIGING_HELP_TEXT = mark_experimental(
+    """filter objects depending on their `IndicatieMachtiging`:
+* `eigen`: Toon objecten waarvan het attribuut `IndicatieMachtiging` leeg is.
+* `gemachtigde`: Toon objecten waarvan het attribuut `IndicatieMachtiging` 'gemachtigde' is.
+* `machtiginggever`: Toon objecten waarvan het attribuut `IndicatieMachtiging` 'machtiginggever'
+"""
+)
+
+
+class MachtigingChoices(models.TextChoices):
+    eigen = "eigen", _("Eigen")
+    gemachtigde = "gemachtigde", _("Gemachtigde")
+    machtiginggever = "machtiginggever", _("Machtiginggever")
+
+
+def machtiging_filter(queryset, name, value: str):
+    if value == MachtigingChoices.eigen:
+        return queryset.filter(**{name: ""})
+
+    return queryset.filter(**{name: value})
+
+
+class MaxLoAFilter(filters.ChoiceFilter):
+    def __init__(self, *args, **kwargs):
+        kwargs.setdefault(
+            "choices",
+            DigiDLevelOfAssurance.choices + eHerkenningLevelOfAssurance.choices,
+        )
+        kwargs.setdefault("lookup_expr", "lte")
+
+        # add choice description
+        help_text = kwargs.get("help_text", "")
+        help_text += (
+            "\n \n **Digid:** \n"
+            + build_choice_description_list(DigiDLevelOfAssurance.choices)
+            + "\n \n **eHerkenning:** \n"
+            + build_choice_description_list(eHerkenningLevelOfAssurance.choices)
+        )
+        kwargs["help_text"] = help_text
+
+        super().__init__(*args, **kwargs)
+
+        # rewrite the field_name correctly
+        self._field_name = self.field_name
+        self.field_name = f"_{self._field_name}_order"
+
+    def filter(self, qs, value):
+        if value in filters.EMPTY_VALUES:
+            return qs
+
+        choices = (
+            DigiDLevelOfAssurance
+            if value in DigiDLevelOfAssurance
+            else eHerkenningLevelOfAssurance
+        )
+
+        order_expression = choices.get_order_expression(self._field_name)
+        numeric_value = choices.get_choice_order(value)
```

Comment:
As in have something like:

```
order_expression = (
    self.DigiDLevelOfAssurance_order_expression
    if value in DigiDLevelOfAssurance
    else self.eHerkenningLevelOfAssurance_order_expression
)
```
or 
```
self.order_expression = {
    DigiDLevelOfAssurance:  DigiDLevelOfAssurance.get_order_expression(self._field_name),
    eHerkenningLevelOfAssurance:  eHerkenningLevelOfAssurance.get_order_expression(self._field_name),
}

order_expression = self.order_expression[choices]
```

Reply:
If the values are always different, you could do value:

```python

def __init__(self, *args, **kwargs):
    self.filer_values = dict()

    for value in DigiDLevelOfAssurance:
        self.filer_values[value] = { # or just a list
            "order_expression": DigiDLevelOfAssurance.get_order_expression(self._field_name),
            "numeric_value": DigiDLevelOfAssurance.get_choice_order(value)
        }

    for value in eHerkenningLevelOfAssurance:
        self.filer_values[value] = {
            "order_expression": eHerkenningLevelOfAssurance.get_order_expression(self._field_name),
            "numeric_value": eHerkenningLevelOfAssurance.get_choice_order(value)
        }


def filter(self, qs, value):
    order_expression = self.filer_values[value]["order_expression"]  # make a list and change to [0]
    numeric_value = self.filer_values[value]["numeric_value"]  # make a list and change to [1]

```

But it currently works fine and these are just optimisations that I think might make sense (if values don't change, calculate once).

Reply:
I think this version is a little less readable when a simple approach. It would make more sense if we wanted to create some abstract filter class which could have values from several enums. But I don't think we will have such cases in the near future 

My opinion about readability is subjective, so it's a matter of personal preferences I guess.